httpcore5/src/main/java/org/apache/hc/core5/http/ContentType.java

```diff
@@ -67,105 +68,125 @@ public final class ContentType implements Serializable {
      * @since 5.1
      */
     public static final ContentType APPLICATION_PROBLEM_XML = create(
-            "application/problem+xml", StandardCharsets.UTF_8);
+            "application/problem+xml", StandardCharsets.UTF_8, false);
 
     /**
      * Public constant media type for {@code application/rss+xml}.
      * @since 5.1
      */
     public static final ContentType APPLICATION_RSS_XML = create(
-            "application/rss+xml", StandardCharsets.UTF_8);
+            "application/rss+xml", StandardCharsets.UTF_8, false);
 
     public static final ContentType IMAGE_BMP = create(
-            "image/bmp");
+            "image/bmp", (Charset) null, true);
     public static final ContentType IMAGE_GIF = create(
-            "image/gif");
+            "image/gif", (Charset) null, true);
     public static final ContentType IMAGE_JPEG = create(
-            "image/jpeg");
+            "image/jpeg", (Charset) null, true);
     public static final ContentType IMAGE_PNG = create(
-            "image/png");
+            "image/png", (Charset) null, true);
     public static final ContentType IMAGE_SVG = create(
-            "image/svg+xml");
+            "image/svg+xml", (Charset) null, false);
     public static final ContentType IMAGE_TIFF = create(
-            "image/tiff");
+            "image/tiff", (Charset) null, true);
     public static final ContentType IMAGE_WEBP = create(
-            "image/webp");
+            "image/webp", (Charset) null, true);
     public static final ContentType MULTIPART_FORM_DATA = create(
-            "multipart/form-data", StandardCharsets.ISO_8859_1);
+            "multipart/form-data", StandardCharsets.ISO_8859_1, true);
 
     /**
      * Public constant media type for {@code multipart/mixed}.
      * @since 5.1
      */
     public static final ContentType MULTIPART_MIXED = create(
-            "multipart/mixed", StandardCharsets.ISO_8859_1);
+            "multipart/mixed", StandardCharsets.ISO_8859_1, true);
 
     /**
      * Public constant media type for {@code multipart/related}.
      * @since 5.1
      */
     public static final ContentType MULTIPART_RELATED = create(
-            "multipart/related", StandardCharsets.ISO_8859_1);
+            "multipart/related", StandardCharsets.ISO_8859_1, true);
 
     public static final ContentType TEXT_HTML = create(
-            "text/html", StandardCharsets.UTF_8);
+            "text/html", StandardCharsets.UTF_8, true);
 
     /**
      * Public constant media type for {@code text/markdown}.
      * @since 5.1
      */
     public static final ContentType TEXT_MARKDOWN = create(
-            "text/markdown", StandardCharsets.UTF_8);
+            "text/markdown", StandardCharsets.UTF_8, false);
 
     public static final ContentType TEXT_PLAIN = create(
-            "text/plain", StandardCharsets.UTF_8);
+            "text/plain", StandardCharsets.UTF_8, true);
```

Comment:
Are you certain about this or is this our asssumption these days?

Reply:
According to RFC 7231  text/plain without a specified charset does not have a default anymore, and it's up to the media type definition. However, it's widely accepted today that UTF-8 is the most common and preferred encoding, especially with its prominence in web standards and internationalization efforts.

Reply:
I would really make the charset here explicit, IMHO. No implications.

Reply:
fair enough